src/anml/parameter/spline_variable.py

```diff
@@ -163,7 +163,7 @@ def create_spline(self, df: pd.DataFrame):
             r_linear=self.r_linear
         )
 
-    def _design_matrix(self, df: pd.DataFrame) -> np.ndarray:
+    def _design_matrix(self, df: pd.DataFrame, create_spline: bool = True) -> np.ndarray:
```

Comment:
is `create_spline` used in the function? and sorry, could you remind me of the reason for adding this argument? I remember we had some discussion on checking whether a spline is created or not, and call the function `create_spline` based on that, but not entirely sure the details... >_< thanks!

Reply:
if we pass in a new data frame, we need to create the spline design matrix, but we don't want to create new knots or anything. If we pass `create_spline=False` then it won't overwrite the original knots and bases

Reply:
ah yes, that totally makes sense! thanks! so would we want something like
```
if create_spline:
    self.create_spline(df)
```

Reply:
Ah! Yes thanks for catching that. I just pushed a quick fix to master.